Userland/Libraries/LibCore/ArgsParser.cpp

```diff
@@ -438,12 +438,8 @@ void ArgsParser::add_option(String& value, char const* help_string, char const*
         long_name,
         short_name,
         value_name,
-        [&value](StringView s) {
-            auto value_or_error = String::from_utf8(s);
-            if (value_or_error.is_error())
-                return false;
-
-            value = value_or_error.release_value();
+        [&value](StringView s) -> ErrorOr<bool> {
+            value = TRY(String::from_utf8(s));
```

Comment:
I don't like this, the program should not abort because it was provided non-utf-8 input, it should gracefully error out, maybe print a helpful message (`warnln("Error: {}", that_error); return false;`).

Reply:
Possibly, but I imagine `::from_utf8` can fail because it is unable to allocate _and_ it can fail because the input is not valid utf8?

If that is the case would it make sense to do something like (pseudocode)
```cpp
auto value_or_error = String::from_utf8(s);
if value_or_error.is_error() {
    if value_or_error.error().kind() == NotValidUtf8 {
        warnln("Error: the value `{}` is not valid utf8", s);
        return false;
    }
    return value_or_error.error();
}
return true;
```